src/util/std_types.h

```diff
@@ -1402,6 +1403,16 @@ inline pointer_typet &to_pointer_type(typet &type)
   return static_cast<pointer_typet &>(type);
 }
 
+template<> inline bool can_cast_type<pointer_typet>(const typet &type)
+{
+  return type.id()==ID_pointer && !type.get(ID_width).empty();
```

Comment:
The latter part of this looks like a validation - can we have something that has `type.id()==ID_pointer` without also having an `ID_width` field?